### PR TITLE
opt: faster to_lower by simd

### PR DESCRIPTION
**Pull request by xumingyukou**

### CODE REVIEW

**Comment by liuq19** [comment]
This fixed #419

**Comment by AsterDY** [comment]
please attach Benchmark results

File: native/unittest/test_to_lower.c [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#include "../native.c"
+#include <assert.h>
+#include <string.h>
+
+void test_to_lower(const char* input, const char* expect) {
+    char* dst;
```
Comment (liuq19) [inline native/unittest/test_to_lower.c:6]: 这个测试是否运行过

File: native/to_lower.c [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <immintrin.h>
+
+
+// src 是输入的字符串，可能包含大写字符，dst是转换后的小写字符串，dst位置已经预分配长度为len的内存
```
Comment (liuq19) [inline native/to_lower.c:7]: 如果需要注释，需要使用英文

File: native/unittest/test_to_lower.c [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    int len = sizeof(input);
+    char* dst = (char*)malloc(len + 1);
+    to_lower(dst, input, len);
+    assert(strcmp(expect, dst) == 0);
+}
+
+int main() {
+    test_to_lower("Hello, World!", "hello, world!");
+    test_to_lower("12345", "12345");
+    test_to_lower("ABCDEFGHIJKLMNOPQRSTUVWXYZ", "ABCDEFGHIJKLMNOPQRSTUVWXYZ");
```
Comment (liuq19) [inline native/unittest/test_to_lower.c:14]: this test case maybe wrong.

File: native/unittest/test_to_lower.c [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#include "../to_lower.c"
+#include<assert.h>
+
+void test_to_lower(const char* input, const char* expect) {
+    int len = sizeof(input);
+    char* dst = (char*)malloc(len + 1);
+    to_lower(dst, input, len);
+    assert(strcmp(expect, dst) == 0);
```
Comment (liuq19) [inline native/unittest/test_to_lower.c:8]: using strncmp here is better, malloc may not set dst[len] as '\0'.

File: Makefile [hunk 5]
```diff
@@ -30,7 +30,7 @@ TMPL_sse 		:= fastint_amd64_test fastfloat_amd64_test native_amd64_test native_e
 CFLAGS_avx		:= -msse -mssse3 -mno-sse4 -mavx -mpclmul -mno-avx2 -DUSE_AVX=1 -DUSE_AVX2=0
 CFLAGS_avx2		:= -msse -mssse3 -mno-sse4 -mavx -mpclmul -mavx2    -DUSE_AVX=1 -DUSE_AVX2=1 
 CFLAGS_sse		:= -msse -mssse3 -mno-sse4 -mno-avx -mno-avx2 -mpclmul
-TARGETFLAGS		:= -target x86_64-apple-macos11 -nostdlib -fno-builtin -fno-asynchronous-unwind-tables
```
Comment (liuq19) [inline Makefile:33]: pls revert it to old flags

File: native/unittest/test_to_lower.c [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#include "../to_lower.c"
+#include<assert.h>
+
+void test_to_lower(const char* input, const char* expect) {
+    int len = sizeof(input);
```
Comment (liuq19) [inline native/unittest/test_to_lower.c:5]: 这里求长度应该不能用 sizeof, 需要用 strlen

**Review by liuq19 — APPROVED** [review]
thanks, LGTM

**Pull request merged**